basket/base/middleware.py

```diff
@@ -0,0 +1,58 @@
+import contextlib
+import inspect
+import time
+
+from django.conf import settings
+from django.http import Http404
+from django.utils.deprecation import MiddlewareMixin
+
+from basket import metrics
+
+
+class HostnameMiddleware(MiddlewareMixin):
+    """Add header with k8s cluster / pod details for debugging."""
+
+    def process_response(self, request, response):
+        response["X-Backend-Server"] = "/".join(filter(None, [getattr(settings, x) for x in ["CLUSTER_NAME", "K8S_NAMESPACE", "K8S_POD_NAME"]]))
+        return response
+
+
+class MetricsStatusMiddleware(MiddlewareMixin):
+    """Send status code counts to statsd"""
+
+    def process_response(self, request, response):
+        metrics.incr(f"response.{response.status_code}")
+        return response
+
+    def process_exception(self, request, exception):
+        if not isinstance(exception, Http404):
```

Comment:
Do we need an `else` here to capture 404, or is that handled above?

Reply:
`404`s happen here, but there's so many for so many reasons, I think this was written to ignore them in general? This was adapted from the django statsd middleware: https://github.com/django-statsd/django-statsd/blob/master/django_statsd/middleware.py#L33